app/src/main/java/com/edulectronics/tinycircuit/Circuit/CircuitModel.java

```diff
@@ -1,32 +1,59 @@
 package com.edulectronics.tinycircuit.Circuit;
 
+import java.io.Serializable;
+import java.lang.reflect.Array;
 import java.util.ArrayList;
 import java.util.List;
 
 import com.edulectronics.tinycircuit.Models.Components.Component;
+import com.edulectronics.tinycircuit.Models.Components.IComponent;
 import com.edulectronics.tinycircuit.Models.Components.Powersource;
 
 /**
  * Created by Wilmer on 28-11-2016.
  */
 
-public class CircuitModel {
+public class CircuitModel implements Serializable{
 
-    public ArrayList<Component> components;
+    public Component[][] components;
+    private int width, height;
 
-    public CircuitModel(){
-        components = new ArrayList<>();
+    public CircuitModel(int width, int height){
+        components = new Component[width][height];
+        this.width = width;
+        this.height = height;
     }
 
     /*Checks if the Circuit has a connection from the first powersource*/
-    public boolean checkConnection(){
-        for (Component c : components){
-            if (c.getClass() == Powersource.class){
-                if (c.hasOutputConnection()){
-                    return true;
-                }
+    public boolean noComponentsWithoutOutput(){
+        for (Object object: components) {
+            if (object != null) {
+                Component c = (Component) object;
+                if (!c.hasOutputConnection()) { return false; }
```

Comment:
This only needs to be done for the power source! hasOutputConnection() iterates over all components automatically